util/variadic.h

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Common template parameter packs support
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_UTIL_VARIADIC_H
+#define __SYCLCTS_UTIL_VARIADIC_H
+
+#include <cstddef>
+#include <type_traits>
+
+/**
+ * @brief Can be used to run functors with parameter pack and custom parameters
+ * @tparam functorT Functor to use
+ */
+template <class functorT>
+struct run_variadic {
```

Comment:
I have to think a little bit further about what is going on here with all this SFINAE.
Adding an example of use in the comment above, instead of grep-ing for it through the code.
Could it be clearer with some `if constexpr` and `...` interpolation instead of this recursion?

Reply:
Sorry, I'm not quite sure I've got your idea. We are unpacking single parameter into the comma-delimited list of parameters here.
I've provided an usage example using doxygen markdown for code blocks.

Reply:
Discussion continued and resolved below